shared/src/commonMain/kotlin/io/customer/shared/tracking/queue/BackgroundQueue.kt

```diff
@@ -0,0 +1,204 @@
+            TrackingType.EVENT -> Activity.Event(
+                timestamp = dateTimeUtil.nowEpochSeconds,
+                attributes = attributes,
+                name = name,
+            )
+            TrackingType.PAGE -> Activity.Page(
+                timestamp = dateTimeUtil.nowEpochSeconds,
+                attributes = attributes,
+                name = name,
+            )
+            TrackingType.SCREEN -> Activity.Screen(
+                timestamp = dateTimeUtil.nowEpochSeconds,
+                attributes = attributes,
+                name = name,
+            )
+        },
+        listener = listener,
+    )
+
+    override fun queueRegisterDevice(
+        profileIdentifier: String?,
+        device: Device,
+        listener: TaskResultListener<Unit>?,
+    ) = addToQueue(
+        profileIdentifier = profileIdentifier,
+        activity = Activity.AddDevice(
+            timestamp = dateTimeUtil.nowEpochSeconds,
+            device = device.copy(
+                platform = platform.name,
+            ),
+        ),
+        listener = listener,
+    )
+
+    override fun queueDeletePushToken(
+        profileIdentifier: String?,
+        deviceToken: String,
+        listener: TaskResultListener<Unit>?,
+    ) = addToQueue(
+        profileIdentifier = profileIdentifier,
+        activity = Activity.DeleteDevice(
+            device = Device(
+                token = deviceToken,
+                platform = platform.name,
+            ),
+        ),
+        listener = listener,
+    )
+
+    override fun queueTrackMetric(
+        deliveryId: String,
+        deviceToken: String,
+        event: MetricEvent,
+        listener: TaskResultListener<Unit>?,
+    ) = addToQueue(
+        profileIdentifier = null,
+        activity = Activity.Metric(
+            metricEvent = event,
+            deliveryId = deliveryId,
+            deviceToken = deviceToken,
+            timestamp = dateTimeUtil.nowEpochSeconds,
+        ),
+        listener = listener,
+    )
+
+    override fun queueTrackInAppMetric(
+        deliveryId: String,
+        event: MetricEvent,
+        listener: TaskResultListener<Unit>?,
+    ) = addToQueue(
+        profileIdentifier = null,
+        activity = Activity.Metric(
+            metricEvent = event,
+            deliveryId = deliveryId,
+            deviceToken = null,
+            timestamp = dateTimeUtil.nowEpochSeconds,
+        ),
+        listener = listener,
+    )
+
+    private fun addToQueue(
+        profileIdentifier: String?,
+        activity: Activity,
+        listener: TaskResultListener<Unit>? = null,
+    ) = runOnBackground {
+        val result = trackingTaskQueryHelper.insertTask(
+            task = Task(
+                profileIdentifier = profileIdentifier,
+                identityType = workspace.identityType,
+                activity = activity,
+            ),
+        )
+        queueDispatcher.checkForPendingTasks(source = QueueTriggerSource.DATABASE)
```

Comment:
I tried adding listener to database to trigger this automatically, but that didn't work well because we update tasks status often, so we have to request the dispatcher manually. It is working good so far.